cores/esp32/Esp.cpp

```diff
@@ -158,16 +159,48 @@ static uint32_t sketchSize(sketchSize_t response) {
     data.start_addr = running_pos.offset;
     esp_image_verify(ESP_IMAGE_VERIFY, &running_pos, &data);
     if (response) {
-    return running_pos.size - data.image_len;
+        return running_pos.size - data.image_len;
     } else {
-    return data.image_len;
+        return data.image_len;
     }
 }
     
 uint32_t EspClass::getSketchSize () {
     return sketchSize(SKETCH_SIZE_TOTAL);
 }
 
+String EspClass::getSketchMD5()
+{
+    static String result;
+    if (result.length()) {
+        return result;
+    }
+    uint32_t lengthLeft = getSketchSize();
+
+    const esp_partition_t *running = esp_ota_get_running_partition();
+    if (!running) return String();
```

Comment:
please do not use one liners like this :) also please add some `log_e` messages on why the call failed. It will be helpful if something goes wrong.